metricbeat/module/elasticsearch/test_elasticsearch.py

```diff
@@ -34,28 +56,52 @@ def test_metricsets(self, metricset):
         """
         elasticsearch metricset tests
         """
-        es = Elasticsearch(self.get_hosts())
-        self.check_skip(metricset, es)
+        self.check_skip(metricset)
 
-        self.start_trial(es)
         if metricset == "ml_job":
-            self.create_ml_job(es)
+            self.create_ml_job()
         if metricset == "ccr":
-            self.create_ccr_stats(es)
+            self.create_ccr_stats()
 
-        es.indices.create(index='test-index', ignore=400)
         self.check_metricset("elasticsearch", metricset, self.get_hosts(), self.FIELDS +
                              ["service"], extras={"index_recovery.active_only": "false"})
 
-    def create_ml_job(self, es):
-        es_version = self.get_version(es)
-        if es_version["major"] < 7:
-            ml_anomaly_detectors_url = "/_xpack/ml/anomaly_detectors"
-        else:
-            ml_anomaly_detectors_url = "/_ml/anomaly_detectors"
+    def test_xpack(self):
+        """
+        elasticsearch-xpack module tests
+        """
+        es = Elasticsearch(self.get_hosts())
 
+        self.create_ml_job()
+        self.create_ccr_stats()
+
+        self.render_config_template(modules=[{
+            "name": "elasticsearch",
+            "metricsets": [
```

Comment:
Feels like this could get out of date pretty quickly. I wonder if you want to generate this list somehow?

Reply:
One option, although a bit hacky, could be to look for metricset folders that have `data_xpack.go` in them. WDYT?

Reply:
SGTM. Also ok to get it in as is for now and follow up on this improvement.